platforms/software/dependency-management/src/main/java/org/gradle/api/internal/catalog/DefaultVersionCatalogBuilder.java

```diff
@@ -251,7 +253,10 @@ private void maybeImportCatalogs() {
         if (maybeResolvedArtifactResult.isPresent()) {
             ResolvedArtifactResult resolvedArtifactResult = maybeResolvedArtifactResult.get();
             File file = resolvedArtifactResult.getFile();
-            withContext("catalog " + resolvedArtifactResult.getVariant().getOwner(), () -> importCatalogFromFile(file));
+            ComponentIdentifier componentIdentifier = resolvedArtifactResult.getVariants().stream()
+                .map(ResolvedVariantResult::getOwner).findFirst() // There will always be at least one variant
+                .orElseThrow(() -> new IllegalStateException("Artifact selected without variant: " + resolvedArtifactResult.getId()));
```

Comment:
🤔 Having to find the first `ResolvedVariantResult`, that is guaranteed to be there, is a little awkward.  It is also a pattern duplicated in `ChecksumAndSignatureVerificationOverride`.

I wonder if it's worth adding something to `ResolvedArtifactResult` like:

```java
default <T> T variantData(Function<ResolvedVariantResult, T> extractor, String failureMsg) {
    return getVariants().stream()
        .findFirst()
        .map(extractor)
        .orElseThrow(() -> new IllegalStateException("Artifact selected without variant: " + getId()));
}
```

which could then be called like:

```java
resolvedArtifactResult.variantData(ResolvedVariantResult::getOwner);
```